### PR TITLE
add support for ALTER TABLE ... RENAME syntax

### PR DESCRIPTION
**Pull request by tangwz**
#### Fix ALTER TABLE ... RENAME syntax
Issue: pingcap/parser#461

#### MySQL Syntax:

```
alter_specification:
...
  | RENAME [TO|AS] new_tbl_name
...
```

#### Bad SQL Case:

```
ALTER TABLE db.t RENAME = db1.t1
ALTER TABLE t RENAME = t1
```
	
### Check List
	
Tests

- Unit test

### CODE REVIEW

**Comment by tangenta** [comment]
@tangwz Would you please resolve the conflicts?

**Comment by tangwz** [comment]
@tangenta How can I solve the conflicting file parser.go?It's full of confusing numbers.

**Comment by tangenta** [comment]
@tangwz You need to run `make parser` in the root directory of project. `goyacc` will generate a new `parser.go` for you based on `parser.y`.

**Comment by tangwz** [comment]
@tangenta I generated a new parser.go, but it's also conflicting file?

File: ast/ddl.go [hunk 1]
```diff
@@ -1728,6 +1744,10 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 		ctx.WriteKeyWord("INSERT_METHOD ")
 		ctx.WritePlain("= ")
 		ctx.WriteString(n.StrValue)
+	case TableOptionTableCheckSum:
+		ctx.WriteKeyWord("TABLE_CHECKSUM ")
+		ctx.WritePlain("= ")
+		ctx.WritePlainf("%d", n.UintValue)
```
Comment (kennytm) [inline ast/ddl.go]: These two statements can be merged.

```go
		ctx.WritePlainf("= %d", n.UintValue)
```

**Comment by kennytm** [comment]
@tangwz Unfortunately every time another PR is merged before this one, the `parser.go` will need to regenerated to adapt with the latest syntax 😓.

**Comment by tangenta** [comment]
It's a little messy now. You can checkout `master` and apply [these changes](https://github.com/pingcap/parser/pull/489/files/6fb70839c366c51602b1f6e7c6f7f7d6e749fea7) again.

**Comment by tangwz** [comment]
@tangenta PTAL

**Pull request merged**